src/field/pokemon.ts

```diff
@@ -3943,6 +4020,31 @@ export class EnemyPokemon extends Pokemon {
         return { move: moveId, targets: this.getNextTargets(moveId) };
       case AiType.SMART_RANDOM:
       case AiType.SMART:
+        /**
+         * Search this Pokemon's move pool for moves that will KO an opposing target.
+         * If there are any moves that can KO an opponent (i.e. a player Pokemon),
+         * those moves are the only ones considered for selection on this turn.
+         */
+        const koMoves = movePool.filter(pkmnMove => {
+          if (!pkmnMove) {
+            return false;
+          }
+          const move = pkmnMove.getMove()!;
+          const fieldPokemon = this.scene.getField();
+          const moveTargets = getMoveTargets(this, move.id).targets
+            .map(ind => fieldPokemon[ind])
+            .filter(p => this.isPlayer() !== p.isPlayer());
+          // Only considers critical hits for crit-only moves or when this Pokemon is under the effect of Laser Focus
+          const isCritical = move.hasAttr(CritOnlyAttr) || !!this.getTag(BattlerTagType.ALWAYS_CRIT);
```

Comment:
```suggestion
          const isCritical = (move.hasAttr(CritOnlyAttr) || this.getTag(BattlerTagType.ALWAYS_CRIT));
```
I don't think you need to do `!!` here.

Reply:
I tried removing it here, TS got mad at me

Reply:
Hm, alright. Not a big deal then.